migrate tag component to design-system

pvev opened the pull request:
#### Summary

Tag migration component:

- different tags unification
- decoupled from redux (guest tag)
- Alert tag unified to Tag and tooltip support
- re-enabled tests and fixed unit tests in the build branch
- fixed an issue with build due to wrong import plus several other fixes

#### Ticket Link
n/a

#### Screenshots
<img width="1643" height="967" alt="Screenshot 2025-11-06 at 11 27 06 AM" src="https://github.com/user-attachments/assets/27f88686-4d5d-432e-b23b-b60cfce883f0" />
<img width="605" height="761" alt="Screenshot 2025-11-06 at 11 27 21 AM" src="https://github.com/user-attachments/assets/e3c93383-83d0-49bf-b1bb-4ddc156511b7" />
<img width="282" height="674" alt="Screenshot 2025-11-06 at 11 27 29 AM" src="https://github.com/user-attachments/assets/5e965b21-8e0b-4472-a5ec-6d3988cc512d" />
<img width="628" height="543" alt="Screenshot 2025-11-06 at 11 28 06 AM" src="https://github.com/user-attachments/assets/657137f4-b0e0-4466-bb5d-38bb6e61d0be" />
<img width="617" height="281" alt="Screenshot 2025-11-06 at 11 28 28 AM" src="https://github.com/user-attachments/assets/12a11a1f-62a3-4a2d-930f-752ab784999a" />
<img width="761" height="576" alt="Screenshot 2025-11-06 at 11 28 46 AM" src="https://github.com/user-attachments/assets/32bd8b9b-e2f5-4203-9c26-c421e76d5c11" />
<img width="606" height="578" alt="Screenshot 2025-11-06 at 11 29 20 AM" src="https://github.com/user-attachments/assets/5d317ed0-b78e-4b52-b0f4-5439b02afb29" />
<img width="682" height="503" alt="Screenshot 2025-11-06 at 11 29 45 AM" src="https://github.com/user-attachments/assets/b992c836-ab0e-4400-99d6-5788b732d43f" />
<img width="677" height="519" alt="Screenshot 2025-11-06 at 11 31 01 AM" src="https://github.com/user-attachments/assets/b84aaf95-9a41-4485-ba87-88635051569e" />
<img width="724" height="478" alt="Screenshot 2025-11-06 at 11 31 18 AM" src="https://github.com/user-attachments/assets/e49425c5-b56b-46bc-a36b-12edbe63d489" />
<img width="904" height="483" alt="Screenshot 2025-11-06 at 11 31 46 AM" src="https://github.com/user-attachments/assets/5f213392-b62b-4697-a280-64ebd9b09366" />
<img width="668" height="551" alt="Screenshot 2025-11-06 at 11 36 30 AM" src="https://github.com/user-attachments/assets/9e3e74ee-3e2b-4ead-ac16-4fc4e03c79bf" />
<img width="708" height="546" alt="Screenshot 2025-11-06 at 11 37 18 AM" src="https://github.com/user-attachments/assets/a66a9d78-4ed0-4e1a-b903-c1a18ef6be1f" />


#### Release Note
<!--
Add a release note for each of the following conditions:

* Config changes (additions, deletions, updates).
* API additions—new endpoint, new response fields, or newly accepted request parameters.
* Database changes (any).
* Schema migration changes. Use the [Schema Migration Template](https://docs.google.com/document/d/18lD7N32oyMtYjFrJKwsNv8yn6Fe5QtF-eMm8nn0O8tk/edit?usp=sharing) as a starting point to capture these details as release notes. 
* Websocket additions or changes.
* Anything noteworthy to a Mattermost instance administrator (err on the side of over-communicating).
* New features and improvements, including behavioral changes, UI changes, and CLI changes.
* Bug fixes and fixes of previous known issues.
* Deprecation warnings, breaking changes, or compatibility notes.

If no release notes are required, write NONE. Use past-tense. Newlines are stripped.

Examples:

```
Added new API endpoints POST /api/v4/foo, GET api/v4/foo, and GET api/v4/foo/:foo_id.
```

```
Added a new config setting ServiceSettings.FooBar. Added a new column Foo to the Users table.
```

```release-note
NONE
```
-->
```release-note
NONE
```

hmhealey committed:
Add widgets folder

hmhealey committed:
Disable some CI on feature branch

hmhealey committed:
Disable image optimizations to speed up build

hmhealey committed:
Add design-system package

hmhealey committed:
Move channels widget folder

M-ZubairAhmed committed:
Merge branch 'build2025-widgets' of github.com:mattermost/mattermost into build2025-widgets

M-ZubairAhmed committed:
progress_bar

M-ZubairAhmed committed:
Tabs component (#34238)

hmhealey committed:
Fix design system not appearing in autocompletes

matthewbirtch committed:
Button Component (#34233)

* rename threads button and add button component

* fixed lint issues

* update accessibility for loading state

* test updates

* move to platform primitives folder

M-ZubairAhmed committed:
Seperator (#34243)

matthewbirtch committed:
Update button.test.tsx

matthewbirtch committed:
Merge branch 'build2025-widgets' of https://github.com/mattermost/mattermost into build2025-widgets

matthewbirtch committed:
fix issue with folder naming

matthewbirtch committed:
fix issue with button folder name

matthewbirtch committed:
fixed snapshot on thread_item tests

matthewbirtch committed:
Icon Button Component (#34234)

* icon button component and styles

* add tests

* Remove excess prop types from IconButton

* Rename count/countText props and make count a number

* Trim trailing whitespace from icon_button.scss

* remove snapshot

* fix ts errors on test

* move to platform primitives folder

* move to platform primitive folder

* fix lint issues

* comment out withtooltip for now until it's moved

* fix lint error

---------

Co-authored-by: Harrison Healey <harrisonmhealey@gmail.com>

harshilsharma63 committed:
Merge branch 'master' into build2025-widgets

pvev committed:
build2025 - design system - add story book to webapp project and some basic stories (#34249)

saturninoabril committed:
move toggle to design-system (#34242)

* move toggle component to design-system

* moved and bundled scss for toggle

pvev committed:
add missing metadata

hmhealey committed:
Fix Storybook looking for non-existant CSS

matthewbirtch committed:
fix icons variation story for button story

matthewbirtch committed:
Merge branch 'build2025-widgets' of https://github.com/mattermost/mattermost into build2025-widgets

hmhealey committed:
Add design-system package to postinstall script

hmhealey committed:
Update how components are exported from DS package

matthewbirtch committed:
fixed icon button and button story

matthewbirtch committed:
Merge branch 'build2025-widgets' of https://github.com/mattermost/mattermost into build2025-widgets

hmhealey committed:
Fix CSS build for DS package

marianunez committed:
Extract input and withTooltip components (#34258)

* Extrated inputs component to common components

* Linting

* linting

* More linting

* PR Feedback

* Linting

---------

Co-authored-by: Mattermost Build <build@mattermost.com>

hmhealey committed:
Temporarily remove SASS utilities

harshilsharma63 committed:
radio buttons (#34244)

* WIP

* migrated all radio buttons

* WIP

* fixed linter

* removed accidental change

* Fixed jest config for new component

* wip

* wip

* test fix

* test fix

* fixed import path

* fixed import path

* removed debug

M-ZubairAhmed committed:
Scrollbar (#34248)

pvev committed:
add provider to fix redux coupled components

pvev committed:
fix toggle styles

matthewbirtch committed:
fixes to button compnent icon sizes and stories

hmhealey committed:
Merge remote-tracking branch 'origin/master' into build2025-widgets

hmhealey committed:
Update version of DS package to 11.1.0 and add DS package to update-versions script

matthewbirtch committed:
Build2025 storybook theme support (#34293)

* Add Mattermost theme support to Storybook

- Add theme-utils.ts with theme definitions and CSS variable application
- Update preview.tsx with theme toolbar and decorator
- Add THEMES.md documentation for using themes in Storybook
- Support all 5 Mattermost themes: Denim, Sapphire, Quartz, Indigo, Onyx
- Themes switch dynamically via toolbar paintbrush icon

* Refactor theme-utils to reuse existing webapp utilities

- Remove duplicate toRgbValues and blendColors functions
- Import toRgbValues from utils/utils.tsx
- Import blendColors from mattermost-redux/utils/theme_utils
- Reduce file from 188 to 138 lines by reusing existing code
- Keep only Storybook-specific logic (applyThemeToStorybook)

* Simplify theme-utils to directly use existing applyTheme

- Remove all manual CSS variable setting code (100+ lines)
- Import and use existing applyTheme() from utils/utils.tsx
- applyTheme already handles all theme application logic:
  - CSS custom properties
  - Dynamic changeCss() calls
  - Desktop app integration
  - Code theme updates
- Reduce file from 138 to 59 lines (57% reduction)
- Keep only Storybook-specific config (toolbar options, theme keys)

* Fix background color not applying in Storybook

- Explicitly set backgroundColor and color on .app__body div
- Use theme.centerChannelBg and theme.centerChannelColor inline styles
- Move themeObject to component scope so it updates with theme changes
- Add themeObject to useEffect dependency array for proper updates

* update to themes styles and definition

* Move .app__body class to body element and use built-in centring logic

* Re-enable background colour picker

* Update theming documentation and move it into Storybook fully

---------

Co-authored-by: Harrison Healey <harrisonmhealey@gmail.com>

pvev committed:
migrate storybook to latest version; (#34316)

* migrate storybook to latest version;

* Update package-lock.json

* Upgrade version of Node 20 to support Storybook 10

* Disable Storybook telemetry

* Fix incorrect update to backgrounds while merging

* Update import paths in docs

---------

Co-authored-by: Harrison Healey <harrisonmhealey@gmail.com>

pvev committed:
migrate tag component to design-system

pvev committed:
fix linter

pvev committed:
adjust typescript and renable unit tests

hmhealey committed:
Merge branch 'master' into build2025-widgets

hmhealey commented on webapp/channels/src/selectors/guest_tags.test.ts:
```diff
@@ -0,0 +1,151 @@
+
+            expect(shouldHideGuestTags(state1)).toBe(false);
+            expect(shouldHideGuestTags(state2)).toBe(true);
+        });
```
issue: These tests don't really do anything because the value returned by this selector is a primitive and because this selector doesn't do anything complicated enough to memoize

hmhealey commented on webapp/platform/design-system/src/components/primitives/tag/tag.tsx:
```diff
@@ -0,0 +1,275 @@
+    icon?: ReactNode;
+
+    /** Icon size in pixels - will be auto-calculated based on tag size if not provided */
+    iconSize?: number;
```
Why make this configurable? I think it would be better to minimize the number of additional props that these components need

hmhealey commented on webapp/platform/design-system/src/components/primitives/tag/tag.tsx:
```diff
@@ -0,0 +1,275 @@
+    hide?: boolean;
+
+    /** Full width variant */
+    fullWidth?: boolean;
```
Do we actually need this? I don't see any reference to this in the docs or anywhere that we use it

matthewbirtch replied:
I don't see a need for it. I say remove it

hmhealey commented on webapp/platform/design-system/src/components/primitives/input/input.test.tsx:
Could you revert the changes to the other test files and submit those in a separate PR? I'm going to be cherry-picking these changes onto another branch in the future, and it'd be much easier for me to do that if we keep these PRs specific to one component at a time

pvev replied:
sure, but probably will have to be done in a branch pointing to this one. Or duplicate some changes that are in this branch. I will see, but for now, this branch only contains changes related to Tag

pvev committed:
implement PR feedback

matthewbirtch commented on webapp/channels/src/components/add_user_to_group_multiselect/multiselect_option/multiselect_option.tsx:
```diff
@@ -4,16 +4,16 @@
 import {Client4} from 'mattermost-redux/client';
 import {isGuest} from 'mattermost-redux/utils/user_utils';
 
+import GuestTag from 'components/guest_tag';
```
I'm kind of confused by this. why is guest_tag in the components folder and a preset also in the design-system folder

matthewbirtch commented on webapp/channels/src/components/guest_tag/index.tsx:
```diff
@@ -0,0 +1,49 @@
+import {defineMessages} from 'react-intl';
+import {useSelector} from 'react-redux';
+
+import {GuestTag as PureGuestTag} from '@mattermost/design-system';
```
This feels really odd to be importing GuestTag as PureGuestTag inside the index file for the guest_tag component

matthewbirtch commented on webapp/platform/design-system/src/components/primitives/tag/tag.stories.tsx:
```diff
@@ -0,0 +1,423 @@
+    render: () => (
+        <div style={{padding: '20px', background: '#f0f0f0'}}>
+            <Tag text='Default Tag' variant='default' size='xs' />
+        </div>
```
```suggestion
        <Tag text='Default Tag' variant='default' size='xs' />
```

matthewbirtch replied:
Do we need the container with padding and background? I assume we don't and can remove it for all cases below

matthewbirtch commented on webapp/platform/design-system/src/components/primitives/tag/tag.scss:
```diff
@@ -0,0 +1,167 @@
+    &:is(button) {
+        border: none;
+        font: inherit;
+        outline: inherit;
```
```suggestion
```

I don't think we need this actually. It's messing up the font on the tag

matthewbirtch commented on webapp/platform/design-system/src/components/primitives/tag/tag.stories.tsx:
```diff
@@ -0,0 +1,423 @@
+                variant='info'
+                size='sm'
+            />
+        </div>
```
```suggestion
        <div style={{maxWidth: '200px',}}>
            <Tag
                text='This is a very long tag text that will overflow'
                variant='info'
                size='sm'
            />
        </div>
```
don't think we need the padding/border

matthewbirtch commented on webapp/platform/design-system/src/components/primitives/tag/tag.stories.tsx:
```diff
@@ -0,0 +1,423 @@
+            </div>
+        </div>
+    ),
+};
```
```suggestion
```
Remove

matthewbirtch reviewed (changes requested):
Just a few changes and questions from me. Let me know if you need any clarification

hmhealey committed:
Merge branch 'master' into build2025-widgets

hmhealey committed:
Update Jest config to prioritize CSS mocks over importing packages

Jest's `moduleNameMapper` setting matches the first module when matching import
paths, so we need to specifically tell it to use CSS files first because the
compiled files in the DS package use an absolute path for CSS.

hmhealey committed:
Merge branch 'build2025-widgets' into build2025-widgets-move-tag-component

hmhealey committed:
Merge branch 'master' into build2025-widgets

matthewbirtch committed:
Extract Spinner from Button and Icon Button in to it's own component (#34267)

* extract spinner in to its own component to be used by button and icon button

* refactor to use pixel value

* fix masking style

* updated tests

* simplied story

* Update spinner.stories.tsx

* fix lint issue with story file

* revert to version without loading spinners stories

* address feedback about subpixel values

* remove aria-hidden

* Update button.test.tsx

---------

Co-authored-by: Harrison Healey <harrisonmhealey@gmail.com>

hmhealey committed:
Merge branch 'master' into build2025-widgets

hmhealey committed:
Merge branch 'build2025-widgets' into build2025-widgets-move-tag-component

pvev commented:
/update-branch

mattermost-build commented:
Error trying to update the PR.
Please do it manually.

pvev commented:
closing for now to free the pending review list. Once there is bandwidth will be adjusted to start from the correct base and continue from where was left including the review comments.
